mm2src/adex_cli/src/activation_scheme_db/activation_scheme_impl.rs

```diff
@@ -0,0 +1,75 @@
+use log::{debug, error, warn};
+use serde_json::{json, Value as Json};
+use std::collections::HashMap;
+
+use super::init_activation_scheme::get_activation_scheme_path;
+use crate::helpers::read_json_file;
+
+pub struct ActivationScheme {
+    scheme: HashMap<String, Json>,
+}
+
+impl ActivationScheme {
+    fn new() -> Self {
+        Self {
+            scheme: HashMap::<String, Json>::new(),
+        }
+    }
+
+    pub fn get_activation_method(&self, coin: &str) -> Option<Json> {
+        let Some(Json::Object(object)) = self.scheme.get(coin) else { return None };
+        let mut copy = json!({});
+        for (k, v) in object.iter() {
+            // WORKAROUND: serde_json::Value does not support removing key
+            if *k == "userpass" {
+                continue;
+            }
+            copy[k] = v.clone();
+        }
+        Some(copy)
+    }
+
+    fn init(&mut self) -> Result<(), ()> {
```

Comment:
This is in order to be able to use `?` instead of compound constructions

Reply:
It certainly is not  ignoring errors. No any errors are ignored, you can check it easy

Reply:
with a quick look I was able to find this one:
https://github.com/KomodoPlatform/atomicDEX-API/blob/58e69ab94a985431e10a0fc4a1a75e5b053de054/mm2src/adex_cli/src/helpers.rs#L21-L23

The caller will never know what was the problem of failing it. It's overally doesn't have the error handling mechanism. And `(), ()` is also redundant to have. Just don't return anything at all if there is nothing to return :)

Reply:
> The caller should know nothing except the error occurred in this certain code to stop further processing

You mean generally? Or just for this particular code I shared?

Reply:
It's like I would return bool, but `?` shortcut is a great feature and I would like to keep using  it

Reply:
I think it's wrong and it's as same as not returning anything and calling `std::process::exit` with printing the log.

Reply:
`adex_cli` is a single-pass application that is why you're right. Using Result<(),()> is the same as using `std::process::exit` but in more flexible way. If I didn't return `Result<T, ()>` or `Result<(), ()>` I would have to use ugly constructions like everywhere.

```rust
if let Err(_) = calling_method(...) {
return ...;
}
```
instead of :

```rust
calling_method(...)?;
```

Tell me please why should I give this ability up ). Using this trick I create more readable and concise healthy code, is it really wrong?

Reply:
Thank you, Onur, for your suggestion.
`anyhow` is certainly decision quite certain and quite flexible. 
I have introduced it 
done )